### ISSUE TITLE
Timestamp keys from ZonedDateTime

### ISSUE THREAD

**Issue by zmumi** [FasterXML/jackson-modules-java8#128]
As of v2.9.9, the following fails

```java

class Sample {
    private Map<ZonedDateTime, ZonedDateTime> map;
    private ZonedDateTime dateTime;

    public Sample() {
    }

    public ZonedDateTime getDateTime() {
        return dateTime;
    }

    public void setDateTime(ZonedDateTime dateTime) {
        this.dateTime = dateTime;
    }

    public Map<ZonedDateTime, ZonedDateTime> getMap() {
        return map;
    }

    public void setMap(Map<ZonedDateTime, ZonedDateTime> map) {
        this.map = map;
    }
}

public class JacksonTest {

    @Test
    public void shouldSerializeAsTimestampInMapKeys() throws IOException {
        // given
        final ObjectMapper objectMapper = new ObjectMapper();
        objectMapper.registerModule(new Jdk8Module());
        objectMapper.registerModule(new JavaTimeModule());
        objectMapper.enable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
        objectMapper.enable(SerializationFeature.WRITE_DATE_KEYS_AS_TIMESTAMPS);

        final ZonedDateTime datetime = ZonedDateTime.parse("2007-12-03T10:15:30+01:00[Europe/Warsaw]");
        final HashMap<ZonedDateTime, ZonedDateTime> map = new HashMap<>();
        map.put(datetime, datetime);
        final Sample sample = new Sample();
        sample.setMap(map);
        sample.setDateTime(datetime);

        // when
        final String value = objectMapper.writeValueAsString(sample);

        // then
        assertThat(value).isEqualTo("{\"map\":{\"1196673330.000000000\":1196673330.000000000},\"dateTime\":1196673330.000000000}");
    }
}
```

Option `SerializationFeature.WRITE_DATE_KEYS_AS_TIMESTAMPS` has no impact on ZonedDateTime. Maybe this option should work in this configuration? (I see in docs it only works for `java.util.Date`).

**Comment by cowtowncoder** [comment]
Sounds like it should work with that type too.

I do not have time to work on this in near term myself, but could help getting a PR merged for 2.10 if you (or anyone else) has time.

**Comment by cowtowncoder** [comment]
Seems reasonable, and I think this would be relatively easy.

If implemented, needs to be in new minor version (since it changes observed behavior, not just simple bug fix): as of now that would be 2.10 (or 3.0, `master`).
